designs/discontinue_usage_of_kube_rbac_proxy.md

```diff
@@ -0,0 +1,281 @@
+>
+>Also, there was concern in the past because AWS did not support it. However, this changed,
+>as detailed in their announcement: [Amazon VPC CNI now supports Kubernetes Network Policies](https://aws.amazon.com/blogs/containers/amazon-vpc-cni-now-supports-kubernetes-network-policies/).
+>
+>Moreover, under this proposal users still able to disable/enable this option as please them.
+
+1. NetworkPolicy is a simple firewall and does not provide authn/authz and encryption.
+
+> Yes, that's correct. NetworkPolicy acts as a basic firewall for pods within a Kubernetes cluster, controlling traffic flow at the IP address or port level. However, it doesn't handle authentication (authn), authorization (authz), or encryption directly likw kube-rbac-proxy solution.
+
+1. Could not Kubebuilder maintainers use the shared infrastructure to continue building and promoting those images under the new `registry.k8s.io`?
+
+> We tried to do that, see [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/k8s-staging-kubebuilder.yaml) the recipe implemented.
+> However, it does not work because kube-rbac-proxy is not under the
+> kubernetes umbrella. More over we experiment the GitHub Repository as alternative approach, see the [PR](https://github.com/kubernetes-sigs/kubebuilder/pull/3854) but seems
+> that we are not allowed to use it. Nevertheless, both approaches does not show to sort out all motivations and requirements
+> Ideally, Kubebuilder should not be responsible for maintain and promote third-part artifacts.
+
+1. However, is not Kubebuilder also building and promoting the binaries required to be used within [EnvTest](./../docs/book/src/reference/envtest.md)
+feature implemented in controller-runtime?
+
+> Yes, but it also will need to change. Controller-runtime maintainers are looking for solutions to
+> built those binaries inside its project since it seems part of its domain. This change is likely
+> to be transparent to the community users.
+
+1. Could we not use the Controller-Runtime feature [controller-runtime][cr-pr] which enable secure metrics serving over HTTPS?
+
+No. After we ask for a hand for reviews from skilled auth maintainers and receiving feedback, it appears that this configuration does not
+align with best practices. Currently, options such as cache TTL, anonymity, and timeouts are not configurable,
+and there is no way to override them. Moreover, the certificate authentication seems to be broken.
+The `alwaysallow` option for certain paths (like `healthz`) is not configurable, and the `alwaysallowgroups` (e.g., `system:masters`)
+does not seem to be properly configured. This configuration is likely to cause issues with vendors like OpenShift,
+which provide Prometheus for the entire cluster. See how this vendor attempts to overwrite some configurations to
+work properly within their environment: [OpenShift kube-rbac-proxy pull request #43](https://github.com/openshift/kube-rbac-proxy/pull/43/files).
```

Comment:
not openshift specific.  See https://github.com/brancz/kube-rbac-proxy/pull/125.

Reply:
I will update it here. Thank you for the help.

Reply:
Closed since it is addressed in the proposal.
We are linking the issue open to Controller-Runtime